### ISSUE TITLE
Wrong use of str::rsplit in case_sensitive_file_extension_comparisons example

### ISSUE THREAD

**Issue by kadiwa4** [rust-lang/rust-clippy#9220]
This is the current example for `case_sensitive_file_extension_comparisons`:
```rust
fn is_rust_file(filename: &str) -> bool {
    filename.ends_with(".rs")
}
```
Use instead:
```rust
fn is_rust_file(filename: &str) -> bool {
    filename.rsplit('.').next().map(|ext| ext.eq_ignore_ascii_case("rs")) == Some(true)
}
```

The suggested replacement is wrong and misleading. `filename.rsplit('.').next()` will _always_ return `Some(_)`. If there is no `.` in filename, this function will instead check if the entire input string is equal to `"rs"`. Thus, this returns `true`: `is_rust_file("rs")`

My suggested replacement:
```rust
fn is_rust_file(filename: &str) -> bool {
    filename.rsplit_once('.').map_or(false, |(_, ext)| ext.eq_ignore_ascii_case("rs"))
}
```
Unfortunately, `str::rsplit_once` was only added in version 1.52.0.

I can’t file a PR myself at the moment.

**Comment by flip1995** [comment]
I don't think `split` should be used here, but rather `Path`, like it was suggested in the original issue:

```rust
let filename = Path::new(filename);
if filename.extension().map(|ext| ext.eq_ignore_ascii_case("abc")) != Some(true) {
    panic!("Please provide an ABC file");
}
```

Both `rsplit_once` and `eq_ignore_ascii_case` will need a MSRV on the lint.

**Comment by FoseFx** [comment]
Fixed via #9229
